pkg/common/cns-lib/vsphere/virtualcentermanager.go

```diff
@@ -183,3 +186,26 @@ func (m *defaultVirtualCenterManager) IsOnlineExtendVolumeSupported(ctx context.
 	log.Infof("Online volume expansion is not supported on vCenter version %q", vCenterVersion)
 	return false, nil
 }
+
+// IsCnsSnapshotSupported checks if cns snapshot is supported or not.
+func (m *defaultVirtualCenterManager) IsCnsSnapshotSupported(ctx context.Context, host string) (bool, error) {
+	log := logger.GetLogger(ctx)
+
+	// Get VC instance.
+	vcenter, err := m.GetVirtualCenter(ctx, host)
+	if err != nil {
+		log.Errorf("Failed to get vCenter. Err: %v", err)
+		return false, err
+	}
+	vcVersion := vcenter.Client.ServiceContent.About.Version
+	isvSphere70U3orAbove, err := IsvSphereVersion70U3orAbove(ctx, vcenter.Client.ServiceContent.About)
```

Comment:
Using CheckAPI may be better. Ignore it if it would introduce cyclic import.

Reply:
Yeah, it creates a circular import.